Frontend/src/Components/Dashboard/GraphContainer.js

```diff
@@ -0,0 +1,253 @@
+import { Select, useConst, VStack } from "@chakra-ui/react";
+import { useCallback, useMemo, useState } from "react";
+import CustomGraph from "../Graph/CustomGraph";
+import GraphData from "../Graph/graph-data.json";
+
+/**
+ * Generates color data for all categories/datasets and packs the data nicely
+ *
+ * The data is in this format:
+ * ```
+ * [
+ * [
+ *   {
+ *     category: string,
+ *     values: [
+ *       {
+ *         key: string,
+ *         name: string,
+ *         color: string
+ *       }
+ *     ]
+ *   }
+ * ],
+ * [
+ *   {
+ *     key: string,
+ *     name: string,
+ *     color: string
+ *   }
+ * ]
+ * ]```
```

Comment:
Why an array of arrays with the second array (lines 24-30) not having any categories for the values? All values should have categories, and if they don't, why not put them under some default category _"foo"_ to keep a more uniform structure?

Reply:
I _could_ but then I would have to explicitly exclude that category on a lot of operations. It is definitely a valid solution, but I just chose this one since it would involve refactoring at the time which was more sensible.